src/R/Components/Impl/Controller/IMenuCommand.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.R.Components.Controller {
+    /// <summary>
+    /// An object that implements a single command. 
+    /// </summary>
+    public interface IMenuCommand {
```

Comment:
My suggestion is to do this gradually, specifically:

- Rename existing interface to `IEditorCommand` or whatever is most appropriate for its actual use.
- Define a new `ICommand` interface with the desired properties (async etc), and start using it.
- Eventually, migrate all command implementations to that new interface, and retire the old one (or shim it).